src/rpy_diffusive_flux.cc

```diff
@@ -0,0 +1,166 @@
+        }
+
+    for (const auto &i : state->getTypes())
+        {
+        const double a_i = 0.5 * diameters_(i);
+        auto rho_i = state->getField(i)->const_view();
+        auto flux_i = fluxes_(i)->view();
+
+        // fill flux with zeros initially
+        std::fill(flux_i.begin(), flux_i.end(), 0.);
+
+        // RPY flux from all species
+        for (const auto &j : state->getTypes())
+            {
+            const double a_j = 0.5 * diameters_(j);
+            auto rho_j = state->getField(j)->const_view();
+            auto mu_ex_j = (excess) ? excess->getDerivative(j)->const_view() : Field::ConstantView();
+            auto V_j = (external) ? external->getDerivative(j)->const_view() : Field::ConstantView();
+            bool no_flux = false;
+            for (int idx = 0; idx < mesh->shape(); ++idx)
+                {
+                if (V_j)
+                    {
+                    bool Vidx_inf = std::isinf(V_j(idx));
+                    if (Vidx_inf)
+                        {
+                        if (V_j(idx) > 0 && rho_j(idx) > 0)
+                            {
+                            // ERROR: can't be V=+inf and have positive rho
+                            }
+                        else if(V_j(idx) < 0)
+                            {
+                            // ERROR: can't have V=-inf ever, it is a mass sink
+                            }
+                        }
+                    no_flux = (Vidx_inf || std::isinf(V_j(idx-1)));
+                    }
+                if (!no_flux)
+                    {
+                    const auto x = mesh->lower_bound(idx);
+                    if (x==0)
+                        {
+                        flux_i(idx) = 0;
+                        }
+                    else
+                        {
+                        auto rho_x = mesh->interpolate(x, rho_i);
+                       
+                        //BD flux calculation
+                        if(i==j)
+                            {
+                            auto dmu_ex = 0.0;
+                            if (mu_ex_j) dmu_ex += mesh->gradient(idx,mu_ex_j);
+                            if (V_j) dmu_ex += mesh->gradient(idx,V_j);
+                            double D = 1/(6*M_PI*viscosity_*a_j);
+                            flux_i(idx) += -D*(mesh->gradient(idx,rho_j) + rho_x*dmu_ex);
+                            }
+                       
+                        //RPY flux calculation   
+                        const double d_ij = a_i + a_j;
+                        
+                        //To remove the concern about the lower bound value spill over the buffer sites
+                        double lower_ig = ceil((x-d_ij-mesh->lower_bound())/mesh->step());
+                        const int ig_low = mesh->bin(std::max(lower_ig, 0.));
+                        const int ig_high = floor((mesh->upper_bound()-(x+d_ij))/mesh->step());
```

Comment:
I don't think you want to do this in this way, won't it lead to rounding up? I think you want to just use the standard bin method to round down:
```suggestion
                        const int ig_high = mesh->bin(x+d_ij);
```